packages/amplify-provider-awscloudformation/src/__tests__/iterative-deployment/stack-event-monitor.test.ts

```diff
@@ -0,0 +1,41 @@
+import { StackEventMonitor, IStackProgressPrinter } from '../../iterative-deployment/stack-event-monitor';
+import { CloudFormation } from 'aws-sdk';
+
+const stackProgressPrinterStub = ({
+  print: jest.fn(),
+  start: jest.fn(),
+  stop: jest.fn(),
+} as unknown) as IStackProgressPrinter;
+
+const cfn = ({
+  describeStackEvents: () => ({
+    promise: () => ({
+      NextToken: undefined,
+    }),
+  }),
+} as unknown) as CloudFormation;
+
+jest.useFakeTimers();
+
+describe('StackEventMonitor', () => {
+  const monitor = new StackEventMonitor(cfn, 'testStackName', stackProgressPrinterStub);
+
+  test('start StackEventMonitor', () => {
+    monitor.start();
+
+    jest.runAllTimers();
+    expect(setTimeout).toHaveBeenCalledTimes(1);
+    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 5000);
+    expect(stackProgressPrinterStub.start).toBeCalled();
+    setTimeout(() => {
+      expect(stackProgressPrinterStub.print).toBeCalled();
+    }, 0); // print is called asynchronously by setTimeout, so we need to queue "expect" to run after all tasks are done
```

Comment:
Does `setImmediate()` work instead of `setTimeout()` here?

Reply:
Yes, it does. I was not aware about `setImmediate()`. Have made the change 👍🏻